src/gen/conditions.rs

```diff
@@ -164,11 +164,16 @@ pub fn parse_args(
             Ok(Condition::CreateCoin(puzzle_hash, amount, a.null()))
         }
         RESERVE_FEE => {
-            let fee = parse_amount(a, first(a, c)?, ErrorCode::ReserveFeeConditionFailed)?;
             if (flags & STRICT_ARGS_COUNT) != 0 {
                 check_nil(a, rest(a, c)?)?;
             }
-            Ok(Condition::ReserveFee(fee))
+            match parse_amount(a, first(a, c)?, ErrorCode::ReserveFeeConditionFailed) {
+                Err(ValidationErr(n, ErrorCode::NegativeAmount)) => {
+                    Err(ValidationErr(n, ErrorCode::ReserveFeeConditionFailed))
```

Comment:
to preserve the existing behavior, you would have to map `AmountExceedsMaximum` to `ReserveFeeConditionFailed` too, right?